src/lambda/vue/samInvokeFrontend.ts

```diff
@@ -0,0 +1,322 @@
+            this.payload = oldState.payload
+        }
+        window.addEventListener('message', ev => {
+            const event = ev.data as SamInvokerResponse
+            switch (event.command) {
+                case 'getRuntimes':
+                    this.runtimes = event.data.runtimes
+                    break
+                case 'getSamplePayload':
+                    this.payload.value = JSON.stringify(JSON.parse(event.data.payload), undefined, 4)
+                    break
+                case 'getTemplate':
+                    this.launchConfig.invokeTarget.target = 'template'
+                    this.launchConfig.invokeTarget.logicalId = event.data.logicalId
+                    this.launchConfig.invokeTarget.templatePath = event.data.template
+                    break
+                case 'loadSamLaunchConfig':
+                    this.clearForm()
+                    this.launchConfig = newLaunchConfig(event.data.launchConfig)
+                    if (event.data.launchConfig.lambda?.payload) {
+                        this.payload.value = JSON.stringify(event.data.launchConfig.lambda.payload.json, undefined, 4)
+                    }
+                    if (event.data.launchConfig.lambda?.environmentVariables) {
+                        this.environmentVariables.value = JSON.stringify(
+                            event.data.launchConfig.lambda?.environmentVariables
+                        )
+                    }
+                    if (event.data.launchConfig.sam?.template?.parameters) {
+                        this.parameters.value = JSON.stringify(event.data.launchConfig.sam?.template?.parameters)
+                    }
+                    if (event.data.launchConfig.api?.headers) {
+                        this.headers.value = JSON.stringify(event.data.launchConfig.api?.headers)
+                    }
+                    if (event.data.launchConfig.api?.stageVariables) {
+                        this.stageVariables.value = JSON.stringify(event.data.launchConfig.api?.stageVariables)
+                    }
+                    if (event.data.launchConfig.sam?.containerBuild === true) {
+                        this.containerBuildStr = 'True'
+                    } else if (event.data.launchConfig.sam?.containerBuild === false) {
+                        this.containerBuildStr = 'False'
+                    }
+                    if (event.data.launchConfig.sam?.skipNewImageCheck === true) {
+                        this.skipNewImageCheckStr = 'True'
+                    } else if (event.data.launchConfig.sam?.skipNewImageCheck === false) {
+                        this.skipNewImageCheckStr = 'False'
+                    }
+                    this.msg = `Loaded config ${event.data.launchConfig.name}`
+                    break
+            }
+        })
+
+        // Send a message back to let the backend know we're ready for messages
+        vscode.postMessage({ command: 'initialized' })
```

Comment:
If this is triggering a function that will set state, do this first so that way we don't overwrite the old state (https://github.com/aws/aws-toolkit-vscode/pull/1995/files#diff-07bfccd7f86d3a30304a05887453738b96a678880a7e6e4e3ef6e07f9dc8824aR94-R98)

Reply:
Ah right forgot that Webviews can save state. Since the 'init' hook is disposed of after the first call this didn't cause any issues, but yeah I should re-arrange stuff for better safety.